hbase-common/src/test/java/org/apache/hadoop/hbase/TestTableName.java

```diff
@@ -172,9 +108,52 @@ public int hashCode() {
     new Names("n2", "table2")
   };
 
+  @Test(expected = IllegalArgumentException.class)
+  public void testInvalidNamespace() {
+    for (String tn : invalidNamespace) {
```

Comment:
I know this isn't your code, but since you're here... this only tests the first value in `invalidNamespace`. What you need is something like

```
for (final String tn : invalidNamespace) {
  Assert.assertThrows(IllegalArgumentException.class, () -> TableName.isLegalFullyQualifiedTableName(Bytes.toBytes(tn)));
}
```

A bunch of these tests appear to make this mistake.

Reply:
Ouch, good catch.